### PR TITLE
Verify yaml

### PR DESCRIPTION
**Pull request by bengl**
Ensures that plugins are tested with the same ranges that they support.

### CODE REVIEW

File: packages/datadog-instrumentations/src/aerospike.js [hunk 1]
```diff
@@ -37,10 +37,23 @@ function wrapProcess (process) {
   }
 }
 
+const versions = (() => {
+  switch (process.versions.node.split('.')[0]) {
```
Comment (tlhunter) [inline packages/datadog-instrumentations/src/aerospike.js:41]:
```js
switch(require('../../versions.js').NODE_MAJOR) {
```

File: .github/workflows/plugins.yml [hunk 2]
```diff
@@ -16,6 +16,10 @@ concurrency:
 
 jobs:
   aerospike-node-16:
+    strategy:
+      matrix:
+        node-version: [16]
```
Comment (Qard) [inline .github/workflows/plugins.yml:21]: Seems like node-version is not actually used?

Comment (bengl) [reply .github/workflows/plugins.yml:21]: Yeah, it's for validation in the script. I'm gonna try to work out a better way of expressing this though.

File: .github/workflows/project.yml [hunk 3]
```diff
@@ -147,3 +147,15 @@ jobs:
       - run: yarn type:test
       - run: yarn type:doc
 
+  verify-yaml:
+    strategy:
+      matrix:
+        node-version: [16, 18, 20, 22]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```
[the remark this answers is not in the document]
Comment (tlhunter) [reply .github/workflows/project.yml:156]: Shouldn't `actions` always be a trusted org? If we can't trust GitHub then why is our code on GitHub?

**Review by tlhunter — DISMISSED** [review]
Seems better than what we have today. That said I'm not sure how a developer can figure out how to fix things based on the resulting error.

File: .github/workflows/project.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (rochdev) [reply .github/workflows/project.yml:153]: Can we just disable this entirely? It just complains about _everything_, making it effectively useless as it just pollutes PRs with dozens of comments like this constantly everywhere.

File: .github/workflows/project.yml [hunk 4]
```diff
@@ -147,3 +147,10 @@ jobs:
       - run: yarn type:test
       - run: yarn type:doc
 
+  verify-yaml:
```
Comment (rochdev) [inline .github/workflows/project.yml:150]: I think this (and the name of the script file too) could use a more descriptive name of what it's doing. Something like `verify-plugin-versions` or anything more descriptive.

**Review by rochdev — DISMISSED** [review]

**Review by Qard — APPROVED** [review]

**Comment by bengl** [comment]
Note that this now removes support (i.e. no longer instruments) Next.js `>=12.0.0 <12.2.0` Since the tests won't run (because `next build` does not successfully run).

**Review by tlhunter — DISMISSED** [review]

**Review by tlhunter — APPROVED** [review]

File: packages/datadog-instrumentations/src/next.js [hunk 5]
```diff
@@ -215,17 +214,17 @@ addHook({
...
 
 addHook({
   name: 'next',
-  versions: DD_MAJOR >= 4 ? ['>=10.2 <11.1'] : ['>=9.5 <11.1'],
+  versions: ['>=10.2 <11.1'],
   file: 'dist/next-server/server/serve-static.js'
 }, serveStatic => shimmer.wrap(serveStatic, 'serveStatic', wrapServeStatic))
 
-addHook({ name: 'next', versions: ['>=11.1'], file: 'dist/server/next-server.js' }, nextServer => {
+addHook({ name: 'next', versions: ['>=11.1 <12.0.0', '>=12.2.0'], file: 'dist/server/next-server.js' }, nextServer => {
```
[the remark this answers is not in the document]
Comment (bengl) [reply packages/datadog-instrumentations/src/next.js:227]: For the versions removed from the range (`12.0.x` and `12.1.x`), `next build` fails, so the tests fail. It doesn't appear that we can test these versions (or that we ever could) so the claim of support was always incorrect.

File: .github/workflows/plugins.yml [hunk 6]
```diff
@@ -15,54 +15,34 @@ concurrency:
...
     strategy:
       matrix:
-        node-version: [18]
-        range: ['5.2.0 - 5.7.0']
+        node-version: [16]
+        range: ['>=4.0.0 <5.2.0']
+        aerospike-image: [ce-5.7.0.15]
+        test-image: [ubuntu-latest]
         include:
+          - node-version: 14 # yes, out of range, but the only way to test the oldest version
```
[the remark this answers is not in the document]
Comment (bengl) [reply .github/workflows/plugins.yml:26]: We'd have to take it out of instrumentation as well, not just the GitHub yaml. Meaning we'd never attempt to to instrument those versions. Is that fine?

**Review by rochdev — APPROVED** [review]

**Pull request merged**